src/Features/Core/Portable/GenerateConstructorFromMembers/AbstractGenerateConstructorFromMembersCodeRefactoringProvider.ConstructorDelegatingCodeAction.cs

```diff
@@ -56,19 +56,25 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
                 var codeGenerationService = languageServices.GetRequiredService<ICodeGenerationService>();
 
                 Contract.ThrowIfNull(_state.DelegatedConstructor);
-                var thisConstructorArguments = factory.CreateArguments(
-                    _state.Parameters.Take(_state.DelegatedConstructor.Parameters.Length).ToImmutableArray());
+                var thisConstructor = _state.DelegatedConstructor.ContainingType == _state.ContainingType;
+                var delegatedParameterNames = _state.DelegatedConstructor.Parameters.Select(p => p.Name).ToImmutableHashSet();
+                var delegatedConstructorArguments = factory.CreateArguments(
+                    _state.Parameters.WhereAsArray(p => delegatedParameterNames.Contains(p.Name)));
 
                 using var _1 = ArrayBuilder<SyntaxNode>.GetInstance(out var nullCheckStatements);
                 using var _2 = ArrayBuilder<SyntaxNode>.GetInstance(out var assignStatements);
 
                 var useThrowExpressions = await _service.PrefersThrowExpressionAsync(_document, _fallbackOptions, cancellationToken).ConfigureAwait(false);
 
-                for (var i = _state.DelegatedConstructor.Parameters.Length; i < _state.Parameters.Length; i++)
+                for (var i = 0; i < _state.Parameters.Length; i++)
                 {
-                    var symbolName = _state.SelectedMembers[i].Name;
                     var parameter = _state.Parameters[i];
 
+                    if (delegatedParameterNames.Contains(parameter.Name))
```

Comment:
overall, this code could use some comments.

Reply:
instead of creating delegatedConstructorArguments  above, it feels like we could just walk teh parameters once in this loop, and add to delegatedConstructorArguments (and continue) if the parameter is in `_state.DelegatedConstructor.Parameters`, right?

Reply:
You're right. Fixed: 6664ea1ba806f528594251e87b2539a6214358cd